### PR TITLE
Feature/286 nonlinear relconnectssurfaceedge

### LINKED ISSUE #286: Nonlinear behavior for RelConnectsSurfaceEdge
### Discussed in https://github.com/StructuralAnalysisFormat/StructuralAnalysisFormat-Doc/discussions/285

<div type='discussions-op-text'>

<sup>Originally posted by **msalak-scia** March 17, 2023</sup>
Similarly to #95 which introduced Nonlinear behavior for hinges on 1D. We would like to extend hinges on 2D the same way. This means adding the possibility to describe nonlinear behavior by assigning a NonlinearFunction to the hinge.

Have a look at already implemented RelConnectsStructuralMember which is already in the latest docs here:
https://www.saf.guide/en/latest/supports-and-hinges/relconnectsstructuralmember.html

Existing object: **RelConnectsSurfaceEdge**
Changed rows:

| Property | Example | Definition |
| --- | --- | --- |
| ux | Free, Rigid, Flexible, **Nonlinear** | Translation in X direction. Free - That is it imposes no constraint in the direction. Rigid - The connection in fully rigid in the specified direction. Flexible - The connection is flexible (elastic) in the specified direction. ~~Parameter Flexible can be linear only, non-linearity is not supported.~~ **Nonlinear – The connection’s behavior is set by a function** |
| uy |  |  |
| uz |  |  |
| fix |  |  |
| fiy |  |  |
| fiz |  |  |
| Stiffness X [MN/m] | 100 | Now required for ux = Flexible **or Nonlinear**. For Nonlinear it will be used for linear analysis and it can be used as a starting point for nonlinear analysis.|
| Stiffness Y [MN/m] | | |
| Stiffness Z [MN/m] | | |
| Stiffness Fix [MNm/rad] | | |
| Stiffness Fiy [MNm/rad] | | |
| Stiffness Fiz [MNm/rad] | | |

New rows:
| Property | Example | Definition |
| --- | --- | --- |
| Function ux | Press only | Reference to object NonlinearFunction, applicable if ux = Nonlinear. Nonlinear function must be of type Translation |
| Function uy | | |
| Function uz | | |
| Function fix | Nonlin fun 3 | Reference to object NonlinearFunction, applicable if fix = Nonlinear. Nonlinear function must be of type Rotation |
| Function fiy | | |
| Function fiz | | |</div>

### PR DESCRIPTION
**Pull request by MartinSalak**
This extends the RelConnectsSurfaceEdge with nonlinear behavior of the hinges. Simillarly to the changes previously done on RelConnectsStructuralMember. Also fixes some parts of description in previously changed RelConnectsStructuralMember which I previously forgot.

There are also two unrelated small things
- one typo in StructuralProxyElement
- fixed link to released docs

closes #286

### CODE REVIEW

File: docs/structural-analysis-elements/structuralproxyelement.md
Comment (msalak-scia) [inline docs/structural-analysis-elements/structuralproxyelement.md:1]: unrelated - only fixing mistakes that were there

File: docs/supports-and-hinges/relconnectsstructuralmember.md
Comment (msalak-scia) [inline docs/supports-and-hinges/relconnectsstructuralmember.md:1]: I extended
- description
- requirement for stiffness which I previously forgot to update
- tiny graphical change in notes

File: docs/supports-and-hinges/relconnectssurfaceedge.md
Comment (msalak-scia) [inline docs/supports-and-hinges/relconnectssurfaceedge.md:1]: These are the main changes

File: docs/supports-and-hinges/relconnectssurfaceedge.md [hunk 1]
```diff
@@ -17,21 +17,27 @@ This object is referring to the edge of StructuralSurfaceMember. Only for this o
...
+| Function uy | String | Nonlin fun 1 | yes, if uy = Nonlinear | Reference to object [NonlinearFunction](../supports-and-hinges/nonlinearfunction.md). Nonlinear function must be of type Translation |
+| Function uz | String | Nonlin fun 1 | yes, if uz = Nonlinear | Reference to object [NonlinearFunction](../supports-and-hinges/nonlinearfunction.md). Nonlinear function must be of type Translation |
+| Function fix | String | Nonlin fun 2 | yes, if fix = Nonlinear | Reference to object [NonlinearFunction](../supports-and-hinges/nonlinearfunction.md). Nonlinear function must be of type Rotation |
+| Function fiy | String | Nonlin fun 2 | yes, if fiy = Nonlinear | Reference to object [NonlinearFunction](../supports-and-hinges/nonlinearfunction.md). Nonlinear function must be of type Rotation |
+| Function fiz | String | Nonlin fun 2 | yes, if fiz = Nonlinear | Reference to object [NonlinearFunction](../supports-and-hinges/nonlinearfunction.md). Nonlinear function must be of type Rotation |
```
Comment (dschuermans) [inline docs/supports-and-hinges/relconnectssurfaceedge.md:37]: Something i've overlooked in the previous PR too, the 

> Nonlinear function must be of type Translation
or
> Nonlinear function must be of type Rotation

is something we currently can't easily validate in the SDK...

**Review by dschuermans — CHANGES_REQUESTED** [review]
Fix the typo's :)

File: docs/getting-started/release-notes.md [hunk 2]
```diff
@@ -20,6 +20,9 @@ Current changes from stable version are noted here.
 * [RelConnectsStructuralMember](../supports-and-hinges/relconnectsstructuralmember.md)
    * enums ux, uy, uz, fix, fiy, fiz extended with new value "Nonlinear"
    * new porperties added: Function ux, Function uy, Function, uz, Function fix, Function fiy, Function fiz
+* [RelConnectsSurfaceEdge](../supports-and-hinges/relconnectssurfaceedge.md)
+   * enums ux, uy, uz, fix, fiy, fiz extended with new value "Nonlinear"
+   * new porperties added: Function ux, Function uy, Function, uz, Function fix, Function fiy, Function fiz
```
[the remark this answers is not in the document]
Comment (MartinSalak) [reply docs/getting-started/release-notes.md]: fixed

File: docs/getting-started/release-notes.md [hunk 3]
```diff
@@ -20,6 +20,9 @@ Current changes from stable version are noted here.
 * [RelConnectsStructuralMember](../supports-and-hinges/relconnectsstructuralmember.md)
    * enums ux, uy, uz, fix, fiy, fiz extended with new value "Nonlinear"
    * new porperties added: Function ux, Function uy, Function, uz, Function fix, Function fiy, Function fiz
```
[the remark this answers is not in the document]
Comment (MartinSalak) [reply docs/getting-started/release-notes.md]: fixed

File: docs/structural-analysis-elements/structuralproxyelement.md [hunk 4]
```diff
@@ -2,11 +2,10 @@
 
 **General solids**
 
-StrucutralProxyElements are solid objects described by their boundary representation (faces of objects). The common use for StrucutralProxyElements can be found for complex part of structures (garage ramps for example), which are difficult to describe with [StructuralCurveMember](structuralcurvemember.md) and [StrucuturalSurfaceMemeber](structuralsurfacemember.md) or for massive parts of structures with difficult geometry such as foundations. With StrucuralProxyElement all important parts of the structure can be transferred via SAF format no matter the complexity.
+StrucutralProxyElements are solid objects described by their boundary representation (faces of objects). The common use for StrucutralProxyElements can be found for complex part of structures (garage ramps for example), which are difficult to describe with [StructuralCurveMember](structuralcurvemember.md) and [StructuralSurfaceMember](structuralsurfacemember.md) or for massive parts of structures with difficult geometry such as foundations. With StrucuralProxyElement all important parts of the structure can be transferred via SAF format no matter the complexity.
```
[the remark this answers is not in the document]
Comment (MartinSalak) [reply docs/structural-analysis-elements/structuralproxyelement.md]: There were a lot of them, I hope I got all.

File: docs/supports-and-hinges/relconnectssurfaceedge.md (hunk 1 above)
Comment (dschuermans) [reply docs/supports-and-hinges/relconnectssurfaceedge.md:37]: We'll look at adding some kind of validation to the SDK

**Pull request merged**